src/Database/Query/SelectQuery.php

```diff
@@ -85,6 +85,14 @@ class SelectQuery extends Query implements IteratorAggregate
      */
     protected ?iterable $_results = null;
 
+    /**
+     * Boolean for tracking whether buffered results
+     * are enabled.
+     *
+     * @var bool
+     */
+    protected bool $useBufferedResults = true;
```

Comment:
I think `$bufferResults` is clearer since doesn't just use the existing buffered results.